distributed/protocol/tests/test_collection.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+from distributed.protocol import serialize, deserialize
+import pandas as pd
+import numpy as np
+from dask.dataframe.utils import assert_eq
+
+
+@pytest.mark.parametrize("collection", [tuple, dict])
```

Comment:
I'm not sure I understand your question. Are you asking if we could combine NumPy and Pandas tests together, or what are the four tests you refer to?

Reply:
Yes, the numpy and the pandas tests, also maybe the cuda equivalent tests if it's easy to isolate the cuda imports with `try-except` blocks.


```python
values = [
    (np.arange(50), "dask"),
    (pd.Series([1, 2, 3]), "dask"),
    (None, "pickle"),
]

try:
    import cudf
except ImportError:
    pass
else:
    values.append((cudf.Series([1, 2, 3]), "cuda"))

try:
...
```

Reply:
Doing that for NumPy/Pandas can be done quite, I've pushed a commit for that. However, for CUDA types, this would incur a lot more conditionals (maybe someone has CuPy but no cuDF installed) which will clutter the code even more, and to be fair, I think it's already very cluttered as it is. That said, if you don't mind, I would prefer keeping them as they are now (different tests for CuPy, cuDF and CPU equivalents).

Reply:
`pickle.loads` can call arbitrary Python code.  `msgpack.loads` just produces base Python objects like int/float/str.  Certainly someone could craft malformed msgpack messages, but the worst that would happen is that it would cause an exception.  People can use `pickle.loads` to run arbitrary code, suck up data, delete files, and so on.

Reply:
I'm fine keeping them separate if you prefer.